src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -135,15 +140,16 @@ convertTxOutputs :: [TxOut] -> [(CAddress, CCoin)]
 convertTxOutputs = map (addressToCAddress . txOutAddress &&& mkCCoin . txOutValue)
 
 mkCTx
-    :: Address            -- ^ An address for which transaction info is forming
+    :: (MonadThrow m, MonadTxExtraRead m)
+    => Address            -- ^ An address for which transaction info is forming
     -> ChainDifficulty    -- ^ Current chain difficulty (to get confirmations)
     -> TxHistoryEntry     -- ^ Tx history entry
     -> CTxMeta            -- ^ Transaction metadata
-    -> CTx
-mkCTx addr diff THEntry {..} meta =
-    -- placeholder, this functionality isn't implemented yet
-    let ctFrom = [] in
-    CTx {..}
+    -> m CTx
+mkCTx addr diff THEntry {..} meta = do
+    txExtra <- maybeThrow (Internal "Transaction is not in the DB") =<< getTxExtra _thTxId
+    let ctFrom = convertTxOutputs $ map toaOut $ toList $ teInputOutputs txExtra
+    pure CTx {..}
```

Comment:
We definitely won't leave the code as is since a major part is still lacking: that is, I would need to refactor `Pos.Explorer.*` modules to be compiled whenever _either_ `WITH_EXPLORER` _or_ `WITH_WALLET` are active. I'll look into `TxHistoryEntry`.